### ISSUE TITLE
Review debug documentation

### ISSUE THREAD

**Issue by guusw** [fragcolor-xyz/shards#392]
https://docs.fragcolor.xyz/contribute/code/getting-started/

The contribute docs don't contain much info about how to debug shards and what tools best to use.

Clearly describe the process and tools for debugging on:
- [ ] Windows (mingw)
- [ ] Windows (clang)
- [ ] Ubuntu
- [ ] OSX

**Comment by guusw** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Whole-Team-623c70cfcdeee40011bd03e3/issues/fragcolor-xyz/shards/392?planning-poker) with Zenhub @Kryptos-FR @sinkingsugar

**Comment by sinkingsugar** [comment]
Seriously this is not a priority, this is something trivial that has no step related to shard...

On windows and linux use gdb, just like you'd do with any gcc compiled binary.
On Mac use lldb.
build with `-DCMAKE_BUILD_TYPE=Debug`.

**Comment by sinkingsugar** [comment]
Also as long as you use gnu toolset rust will be debuggable with gdb just fine. So rust is not an excuse.

**Comment by Kryptos-FR** [comment]
> Seriously this is not a priority, this is something trivial that has no step related to Shards...
> 
> On windows and linux use gdb, just like you'd do with any gcc compiled binary. On Mac use lldb. build with `-DCMAKE_BUILD_TYPE=Debug`.
> 
> Anything else is your own mileage and your own setup that has nothing to do with Shards.

I disagree. We do need some internal documentation on the subject. I spent a full day trying to debug shards and couldn't make it work. It was either not hitting the breakpoint, not breaking upon exception or when doing so it was only showing assembly code. And yes I did have the DEBUG build.

Maybe it is evident for you but it isn't for me and I would rather not waste another day on it. Having some reference documentation to rely upon would certainly make it less painful.

**Comment by sinkingsugar** [comment]
And I'm surprised you didn't have a debug setup yet.

**Comment by sinkingsugar** [comment]
Closing as outdated
